### PR TITLE
Shared secret registration moved to an admin API

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

**Comment by hawkowl** [comment]
matrixbot: retest this please

**Comment by hawkowl** [comment]
matrixbot: retest this please

File: tests/10apidoc/01register.pl [hunk 1]
```diff
@@ -240,23 +239,30 @@ sub matrix_v1_register_user_via_secret
    defined $uid or
       croak "Require UID for matrix_register_user_via_secret";
 
-   my $mac = hmac_sha1_hex(
-      join( "\0", $uid, $password, $is_admin ? "admin" : "notadmin" ),
+   $http->do_request_json(
+      method => "GET",
+      uri    => "/r0/admin/register",
+   )->then(sub{
```
Comment (richvdh) [inline tests/10apidoc/01register.pl]: sorry, the whitespace conventions for sytest are... unusual.

this should have a space inside the parens:

`)->then( sub {`

File: tests/10apidoc/01register.pl [hunk 2]
```diff
@@ -240,23 +239,30 @@ sub matrix_v1_register_user_via_secret
...
+      my ( $nonce ) = @_;
+
+      my $mac = hmac_sha1_hex(
+      join("\0", $nonce->{nonce}, $uid, $password, $is_admin ? "admin" : "notadmin" ),
```
Comment (richvdh) [inline tests/10apidoc/01register.pl]: this needs indenting and space inside the parens

File: tests/10apidoc/01register.pl [hunk 3]
```diff
@@ -240,23 +239,30 @@ sub matrix_v1_register_user_via_secret
...
       "reg_secret"
-   );
+      );
 
-   $http->do_request_json(
+      return $http->do_request_json(
       method => "POST",
```
Comment (richvdh) [inline tests/10apidoc/01register.pl]: indent please

File: lib/SyTest/HTTPClient.pm [hunk 4]
```diff
@@ -105,7 +105,7 @@ sub do_request
          my $message = $response->message;
          $message =~ s/\r?\n?$//; # because HTTP::Response doesn't do this
 
-         return Future->fail( "HTTP Request failed (${\$response->code} $message)",
+         return Future->fail( "HTTP Request failed (${\$response->code} $message $uri)",
```
Comment (richvdh) [inline lib/SyTest/HTTPClient.pm]: space inside these parens too please

**Pull request merged**